src/ring_theory/ideal/basic.lean

```diff
@@ -303,6 +305,56 @@ end pi
 
 end ideal
 
+/-- A right ideal in a semiring `R` is an additive submonoid `s` such that
+`b * a ∈ s` whenever `b ∈ s`. If `R` is a ring, then `s` is an additive subgroup.  -/
+@[reducible] def right_ideal (R : Type u) [semiring R] := submodule Rᵐᵒᵖ R
+
+namespace right_ideal
+
+variables [semiring α] (I : right_ideal α) {a b : α}
+
+protected lemma zero_mem : (0 : α) ∈ I := I.zero_mem
+
+protected lemma add_mem : a ∈ I → b ∈ I → a + b ∈ I := I.add_mem
```

Comment:
We should be able to safely delete these lemmas since there should already be an `add_submonoid_class` instance.
```suggestion
```